dinov2/data/datasets/extended.py

```diff
@@ -22,22 +26,30 @@ def get_image_data(self, index: int) -> bytes:
     def get_target(self, index: int) -> Any:
         raise NotImplementedError
 
-    def __getitem__(self, index: int) -> Tuple[Any, Any]:
+    def __getitem__(self, index: int) -> Union[Tuple[Any, Any], torch.Tensor, Image.Image]:
+        img_bytes = self.get_image_data(index)
         try:
-            image_data = self.get_image_data(index)
-            image = ImageDataDecoder(image_data).decode()
+            # have to copy bc stream not writeable
+            image = torch.frombuffer(np.copy(img_bytes), dtype=torch.uint8)
+            image = decode_image(image, ImageReadMode.RGB)
+            image = (image / 255.0).to(torch.float32)
         except Exception as e:
-            # image_data = self.get_image_data(index)
-            image = np.random.randint(0, 255, (224, 224, 3), dtype=np.uint8)
-            image = Image.fromarray(image)
-            # raise RuntimeError(f"can not read image for sample {index}") from e
+            print(e)
+            print("Error: torch.frombuffer failed, trying PIL...", file=sys.stderr)
+            try:
+                image = ImageDataDecoder(img_bytes).decode()
+            except Exception as e:
+                raise RuntimeError(f"can not read image for sample {index}") from e
+
         target = self.get_target(index)
         target = TargetDecoder(target).decode()
 
         if self.transforms is not None:
             image, target = self.transforms(image, target)
+            return image, target
```

Comment:
`return` statement here is the same in `if` and `else`, can be moved outside of the conditional?

Reply:
Good point, done